src/subscription_manager/cache.py

```diff
@@ -489,26 +476,19 @@ def _sync_with_server(self, uep, consumer_uuid, *args, **kwargs):
         if uep.has_capability("combined_reporting"):
             _combined_profile = [
                 {
-                    "content_type": "rpm",
-                    "profile": combined_profile["rpm"]
-                },
-                {
-                    "content_type": "enabled_repos",
-                    "profile": combined_profile["enabled_repos"]
-                },
-                {
-                    "content_type": "modulemd",
-                    "profile": combined_profile["modulemd"]
-                },
+                    "content_type": key,
+                    "profile": value,
+                }
+                for key, value in combined_profile.items()
             ]
             uep.updateCombinedProfile(
                 consumer_uuid,
-                _combined_profile
+                _combined_profile,
             )
-        else:
+        elif "rpm" in combined_profile:
```

Comment:
This change of logic has potential to break functionality. Candlepin server cannot have capability "combined_reporting", but it still can be able to have capability of receiving package profile.